code/modules/mob/holder.dm

```diff
@@ -62,6 +62,7 @@
 	var/obj/item/weapon/holder/H = new holder_type(loc)
 	src.forceMove(H)
 	H.name = name
+	H.icon_state = icon_state
```

Comment:
Ew. I didn't think about others actually. Was kind of worried about mice not having their color icon move over